zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -62,6 +62,19 @@
     settings = new GUI();
   }
 
+  public Paragraph(String paragraphId, Note note, JobListener listener,
+                   NoteInterpreterLoader replLoader) {
+    super(paragraphId, generateId(), listener);
```

Comment:
No, you made the same mistake as I did the first time implementing this PR. 
Indeed, `Paragraph` extends class `Job` and the `generateId()` string passed to super constructor represents the **jobName**. So indeed we must have distinct job name for each paragraph